### ISSUE TITLE
Crash on type

### ISSUE THREAD

**Issue by cglacet** [vscode-restructuredtext/snooty-parser#18]
**Environment**
- [x] Extension version: v143.0.0
- [x] Visual Studio Code version: Version: 1.53.2
- [x] Operating System version: osx 11.2.1
- [x] Python version: Python 3.8.2
- [x] sphinx version: 3.5.1
- [x] A sample of conf.py or its important fragments: included at the end

**How it occurs**

Every time I type a character in a vscode `.rst` file the extension crashes. Syntax highlighting seems to work. The extension use to work before and I don't recall changing anything on my end (local virtual env so I'm actually pretty sure nothing changed). On the other hand I updated vscode.

**Output**

```bash
INFO:snooty.main:Snooty 1.8.7.dev0 starting
INFO:snooty.language_server:Started
WARNING:pyls_jsonrpc.endpoint:Ignoring notification for unknown method workspace/didChangeConfiguration
ERROR:pyls_jsonrpc.endpoint:Failed to handle request 1
Traceback (most recent call last):
  File "/Users/cglacet/.pyenv/versions/kune-documentation/lib/python3.8/site-packages/pyls_jsonrpc/endpoint.py", line 113, in consume
    self._handle_request(message['id'], message['method'], message.get('params'))
  File "/Users/cglacet/.pyenv/versions/kune-documentation/lib/python3.8/site-packages/pyls_jsonrpc/endpoint.py", line 182, in _handle_request
    handler_result = handler(params)
  File "/Users/cglacet/.pyenv/versions/kune-documentation/lib/python3.8/site-packages/pyls_jsonrpc/dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "/Users/cglacet/.pyenv/versions/kune-documentation/lib/python3.8/site-packages/snooty/language_server.py", line 450, in m_text_document__completion
    return self.completions(textDocument.uri, position)
AttributeError: 'dict' object has no attribute 'uri'
[Error - 7:34:42 AM] Request textDocument/completion failed.
  Message: AttributeError: 'dict' object has no attribute 'uri'
  Code: -32602 
[object Object]
```

**Sphinx conf.py**

In my conf file I have 2 patches to correct bugs I had, the rest of it is pretty standard. Here are the two patches:

```python
"""Patching m2r2"""
import m2r2

current_m2r2_setup = m2r2.setup


def patched_m2r2_setup(app):
    try:
        return current_m2r2_setup(app)
    except (AttributeError):
        app.add_source_suffix(".md", "markdown")
        app.add_source_parser(m2r2.M2RParser)
    return dict(
        version=m2r2.__version__,
        parallel_read_safe=True,
        parallel_write_safe=True,
    )


m2r2.setup = patched_m2r2_setup

"""Patching sphinx titles"""
from docutils import nodes
from sphinx.application import Sphinx
from sphinx.environment.collectors.title import TitleCollector

_process_doc = TitleCollector.process_doc


def process_doc(self, app: Sphinx, doctree: nodes.document) -> None:
    if doctree.traverse(nodes.section):
        _process_doc(self, app, doctree)
    else:
        titlenode = nodes.title()
        app.env.titles[app.env.docname] = titlenode
        app.env.longtitles[app.env.docname] = titlenode


TitleCollector.process_doc = process_doc
```

**Comment by lextm** [comment]
Fixed in 1.8.8 release.

You might manually update `snooty-lextudio` Python package to 1.8.8 (or wait till the next release of the VSCode extension to guide you through).

**Comment by bondezbond** [comment]
confirmed upgraded package manually and doesn't crash anymore. Thanks for the quick response.
